Dnd/dndJavascript.js

```diff
@@ -10,44 +19,31 @@ function showSeason(item) {
     return season_number.replace(/\s+/g, '')
   }
   
-  function getSeasonContent() {
-    const season_number = getSelectedSeasonNumber();
+  function getSeasonContent(season_number) {
```

Comment:
you could make this an async function so you can use `await` like above

Reply:
I don't think there is anything to await though, getSeasonContent doesn't make any other function calls anymore

Reply:
I've made getSeasonContent and getSeasonNumber async now

Reply:
I think you're missing some understanding here- an `async` function returns a `Promise<T>` (an asynchronous body of work with a return type `T`).
By making a function `async`, it allows the usage of the `await` keyword, and you can drop the `.then()` call chains.

As an example, consider this:
```js
function getUserId() {
  return getMyUser()
    .then((response) => response.status === 200 ? response.data : throw 'Failed to fetch user')
    .then((user) => user.Id);
}
````

Using an `async` function you can make it much clearer, and in most cases more concise:
```js
function getUserId() {
  const response = await getMyUser();
  if (response.Status !== 200) {
    throw 'Failed to fetch user';
  }

  return response.data.id;
}
```

Reply:
What I was suggesting was:
```js
  function getSeasonContent(season_number) {
    const season_number = getSelectedSeasonNumber();
    const path = "EpisodeData/" + season_number + ".json"
    return fetch(path).then(function(response) {
      return response.json();
    });
  }
```

becomes

```js
  async function getSeasonContent(season_number) {
    const season_number = getSelectedSeasonNumber();
    const path = "EpisodeData/" + season_number + ".json"
    const seasonContent = await fetch(path);
    return seasonContent.json();
  }

Reply:
Okay, thank you. I'll add this now